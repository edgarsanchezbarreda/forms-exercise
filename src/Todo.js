import React from 'react';

export const Todo = ({ task, id = '1', removeTodo }) => {
    const handleDeleteTodo = () => removeTodo(id);

    return (
        <div className='Todo'>
            <li>{task}</li>
            <button onClick={handleDeleteTodo}>X</button>
        </div>
    );
};
