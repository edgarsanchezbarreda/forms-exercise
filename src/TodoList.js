import React, { useState } from 'react';
import { NewTodoForm } from './NewTodoForm';
import { Todo } from './Todo';

export const TodoList = () => {
    const [todos, setTodos] = useState([]);

    const addTodo = newTodo => {
        setTodos(todos => [...todos, newTodo]);
    };
    const removeTodo = id => {
        setTodos(todos => todos.filter(todo => todo.id !== id));
    };

    return (
        <div className='TodoList'>
            <h1 className='TodoList-header'>Todo App</h1>
            <NewTodoForm addTodo={addTodo} />
            <div className='TodoList-todos'>
                <ul>
                    {todos.map(todo => (
                        <Todo
                            key={todo.id}
                            id={todo.id}
                            task={todo.task}
                            removeTodo={removeTodo}
                        />
                    ))}
                </ul>
            </div>
        </div>
    );
};
