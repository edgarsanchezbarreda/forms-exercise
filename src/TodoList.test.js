import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { TodoList } from './TodoList';
import { Todo } from './Todo';

const addTodo = (todoList, task = 'feed the dogs') => {
    const taskInput = todoList.getByLabelText('New Task');
    fireEvent.change(taskInput, { target: { value: task } });
    const submitButton = todoList.getByText('Add Todo');
    fireEvent.click(submitButton);
};

it('Should render without crashing', () => {
    render(<TodoList />);
});

it('Match snapshot', () => {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});

it('Can add a todo', () => {
    const todos = render(<TodoList />);
    addTodo(todos);

    expect(todos.getByLabelText('New Task')).toBeInTheDocument();
    expect(todos.getByLabelText('New Task')).toHaveValue('');
    expect(todos.getByText('Add Todo')).toBeInTheDocument();

    const addedTodo = todos.getByText('feed the dogs');
    expect(addedTodo).toBeInTheDocument();
});

it('Can delete a todo', () => {
    const todos = render(<TodoList />);
    addTodo(todos);

    const removeButton = todos.getByText('X');
    fireEvent.click(removeButton);

    const addedTodo = todos.queryByText('feed the dogs');
    expect(addedTodo).not.toBeInTheDocument();
});
