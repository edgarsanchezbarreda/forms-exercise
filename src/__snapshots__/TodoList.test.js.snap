// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <h1
      class="TodoList-header"
    >
      Todo App
    </h1>
    <form>
      <label
        for="task"
      >
        New Task
      </label>
      <input
        id="task"
        name="task"
        placeholder="todo"
        type="text"
        value=""
      />
      <button>
        Add Todo
      </button>
    </form>
    <div
      class="TodoList-todos"
    >
      <ul />
    </div>
  </div>
</DocumentFragment>
`;
